soroban-env-host/src/host.rs

```diff
@@ -955,7 +1027,7 @@ impl CheckedEnv for Host {
 
     fn map_has(&self, m: Object, k: RawVal) -> Result<RawVal, HostError> {
         let k = self.associate_raw_val(k);
-        self.visit_obj(m, move |hm: &HostMap| Ok(hm.contains_key(&k).into()))
+        self.visit_obj(m, move |hm: &HostMap| Ok(hm.contains_key(&k)?.into()))
     }
 
     fn map_prev_key(&self, m: Object, k: RawVal) -> Result<RawVal, HostError> {
```

Comment:
This was the original [PR comment](https://github.com/stellar/rs-soroban-env/pull/216#discussion_r921746595) of me addressing @leighmcculloch who had the same question. 
Short answer is, I couldn't find out a better way. The `get_prev`/`get_next` in `OrdMap` would return the same key if it exists, rather than returning the prev/next of it, so I had to delete the key and call it again.